# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/WeakMap.cpp
# Commit: f7b553bd9c0b
# Full Hash: f7b553bd9c0b77b0130d377f9a9c696c8c611a01
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-05-11 21:39:06
# Regressor Bug: 1694538
# File Overlap Count: 1
# Description:
#   Bug 1694538 - Convert gcWeakKeys from <weakmap, key> tuples that require an extra lookup, to <ptr to color, target> tuples that simplify the code substantially r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106226
# ==============================================================================

diff -r 89f7028049ed -r f7b553bd9c0b js/src/gc/WeakMap.cpp
--- a/js/src/gc/WeakMap.cpp	Tue May 11 14:01:20 2021 +0000
+++ b/js/src/gc/WeakMap.cpp	Tue May 11 14:18:38 2021 +0000
@@ -34,10 +34,10 @@
 
 void WeakMapBase::unmarkZone(JS::Zone* zone) {
   AutoEnterOOMUnsafeRegion oomUnsafe;
-  if (!zone->gcWeakKeys().clear()) {
-    oomUnsafe.crash("clearing weak keys table");
+  if (!zone->gcEphemeronEdges().clear()) {
+    oomUnsafe.crash("clearing ephemeron edges table");
   }
-  MOZ_ASSERT(zone->gcNurseryWeakKeys().count() == 0);
+  MOZ_ASSERT(zone->gcNurseryEphemeronEdges().count() == 0);
 
   for (WeakMapBase* m : zone->gcWeakMapList()) {
     m->mapColor = CellColor::White;